src/stdlib.cpp

```diff
@@ -141,4 +145,68 @@ auto next(const CallContext& ctx) -> Vallist {
             "bad argument #1 to 'next' (table expected, got " + a.type() + ")");
     }
 }
+
+auto select(const CallContext& ctx) -> Vallist {
+    auto index = ctx.arguments().get(0);
+    std::vector<Value> args;
+
+    for (auto a = ++ctx.arguments().begin(); a != ctx.arguments().end(); a++) {
+        args.push_back(*a);
+    }
+
+    return std::visit(
+        overloaded{
+            [&args](Number n) -> Vallist {
+                if (n == -1) {
+                    return Vallist({*(--args.end())});
+                } else if (n == 0) {
+                    throw std::runtime_error("bad argument #1 to 'select' (index out of range)");
+                } else {
+                    std::vector<Value> returns;
+                    int i = 1;
+                    for (auto a = args.begin(); a != args.end(); a++, i++) {
+                        if (i < n) {
+                            continue;
+                        } else {
+                            returns.push_back(*a);
+                        }
+                    }
+
+                    return Vallist(returns);
+                }
+            },
+            [&args](String s) -> Vallist {
+                if (std::move(s) == "#") {
+                    int size = args.size();
+                    return Vallist({Value(Number(size))});
+                } else {
+                    throw std::runtime_error(
+                        "bad argument #1 to 'select' (number expected, got string)");
+                }
+            },
+            [](auto a) -> Vallist {
+                throw std::runtime_error(
+                    "bad argument #1 to 'select' (number expected, got " + std::string(a.TYPE) +
+                    ")");
+            }},
+        index.raw());
+}
+
+void print(const CallContext& ctx) {
+    auto* const stdout = ctx.environment().get_stdout();
+    std::string gap;
+
+    for (const auto& arg : ctx.arguments()) {
+        Environment env;
+        CallContext tmp(&env);
+        tmp = tmp.make_new(Vallist({arg}));
+
+        const Value v = to_string(tmp);
```

Comment:
Please use the CallContext from the function argument.

Also `make_new` is templated so you can just specify the arguments directly.

```suggestion
        const Value v = to_string(ctx.make_new(arg));
```